source/dofs/dof_handler_policy.cc

```diff
@@ -229,6 +229,22 @@ namespace internal
       {
         /* -------------- distribute_dofs functionality ------------- */
 
+        template <int dim, int spacedim>
+        inline types::global_dof_index
+        distribute_non_local_dofs_on_cell(
+          const DoFHandler<dim, spacedim> &dof_handler,
+          const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+          types::global_dof_index next_free_dof)
+        {
+          const auto non_local_ids =
+            dof_handler.get_fe().get_non_local_dofs_on_cell(cell);
+          for (auto id : non_local_ids)
+            if (dof_handler.non_local_dofs[id] != numbers::invalid_dof_index)
+              dof_handler.non_local_dofs[id] = next_free_dof++;
```

Comment:
And here, you are doing this once per cell, but because of line 242, it really only happens once over all cells. In reality, however, it happens once per parallel process. This isn't going to work in parallel -- you'll have to define what the ownership is in parallel.

Reply:
I'm not sure I understand your comment here. This is going to set the global index only if it was not set elsewhere. Same logic of the vertex_dofs.